orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -1179,11 +1194,65 @@ common::Status GenerateSubgraph(Graph& graph, Node* start_node) {
   return graph.Resolve();
 }
 
+Status RemoveNonDifferentiableEdgeInPartition(
+    Graph& graph,
+    const std::vector<std::string>& non_differentiable_edge_names) {
+
+  assert(!non_differentiable_edge_names.empty());
```

Comment:
Would it simplify the API here to have this method be able to accept an empty vector (for cases where there are no non-diff edges), and then skip the loop below when there is no work?  [ I may be missing something below, but right now I think we will always call RemoveNonDifferentiableEdgeInPartition in any case, including where we do not have any non-diff edges? ]